v2/internal/pkg/imagebuilder/rebuild_catalog.go

```diff
@@ -0,0 +1,314 @@
+		case len(v.OperatorFilter.TargetCatalog) == 0 && len(v.OperatorFilter.TargetTag) > 0:
+			localDest = dockerProtocol + strings.Join([]string{o.LocalFQDN, imgSpec.PathComponent}, "/") + ":" + v.OperatorFilter.TargetTag
+			remoteDest = strings.Join([]string{o.Destination, imgSpec.PathComponent}, "/") + ":" + v.OperatorFilter.TargetTag
+		case len(v.OperatorFilter.TargetCatalog) == 0 && len(v.OperatorFilter.TargetTag) == 0:
+			localDest = dockerProtocol + strings.Join([]string{o.LocalFQDN, imgSpec.PathComponent}, "/") + ":" + imgSpec.Tag
+			remoteDest = strings.Join([]string{o.Destination, imgSpec.PathComponent}, "/") + ":" + imgSpec.Tag
+		}
+
+		// this is safe as we know that there is a docker:// prefix
+		updatedDest := strings.Split(localDest, dockerProtocol)[1]
+
+		buildOptions, err := getStandardBuildOptions(updatedDest, o.SrcTlsVerify)
+		if err != nil {
+			return result, excludeCatalogs, err
+		}
+
+		buildOptions.DefaultMountsFilePath = ""
+
+		o.Logger.Trace("containerfile %s", string(contents.Bytes()))
+
+		buildStoreOptions, err := storage.DefaultStoreOptions()
+		if err != nil {
+			return result, excludeCatalogs, err
+		}
+
+		buildStore, err := storage.GetStore(buildStoreOptions)
+		if err != nil {
+			return result, excludeCatalogs, err
+		}
+		defer buildStore.Shutdown(false)
+
+		os.MkdirAll("configs", 0644)
+		filecopy.Copy(v.FilteredConfigPath, "./configs")
+		defer os.RemoveAll("configs")
+
+		id, ref, err := imagebuildah.BuildDockerfiles(ctx, buildStore, buildOptions, []string{containerfilePath}...)
+		if err == nil && buildOptions.Manifest != "" {
+			o.Logger.Info("✅ successfully created catalog")
+			o.Logger.Debug("  manifest list id : %s", id)
+			o.Logger.Debug("  image reference  : %s", ref.String())
+		}
+		if err != nil {
+			return result, excludeCatalogs, err
+		}
+
+		var retries *uint
+		retries = new(uint)
+		*retries = 3
+
+		manifestPushOptions := manifests.PushOptions{
+			Store:                  buildStore,
+			SystemContext:          newSystemContext(o.DestTlsVerify),
+			ImageListSelection:     cp.CopyAllImages,
+			Instances:              nil,
+			RemoveSignatures:       true,
+			SignBy:                 "",
+			ManifestType:           "application/vnd.oci.image.manifest.v1+json",
+			AddCompression:         []string{},
+			ForceCompressionFormat: false,
+			MaxRetries:             retries,
+		}
+
+		destImageRef, err := alltransports.ParseImageName(localDest)
+		if err != nil {
+			return result, excludeCatalogs, err
+		}
+
+		_, list, err := manifests.LoadFromImage(buildStore, id)
+		if err != nil {
+			return result, excludeCatalogs, err
+		}
+
+		o.Logger.Debug("local cache destination (rebuilt-catalog) %s", localDest)
+		o.Logger.Debug("destination image reference %v", destImageRef)
+		o.Logger.Debug("pushing manifest list to remote registry")
+		// push the manifest list to local cache
+		_, digest, err := list.Push(ctx, destImageRef, manifestPushOptions)
```

Comment:
It would be better to separate the rebuild of catalog from the push to follow the single-responsibility principle. 

The rebuild of the catalog could only generate the image and save it in the working-dir and then in the batch we push it as we do basically for almost all images (graphimage could be changed to follow the same approach in a separate PR).